src/util/git_repos.rs

```diff
@@ -104,40 +104,23 @@ impl GitRepoManager {
     }
 
     /// Runs git fetch on the cached repo for the software specified by `software_id`, determines
-    /// whether `commit_or_tag` is a commit or tag, and if it is a commit, returns the commit and
-    /// the tags if the commit has any, and if it is a tag, returns the commit for that tag and any
-    /// tags for that commit.  If it's neither, returns Error::NotFound.  Returns an error for any
-    /// other failures
-    pub fn get_commit_and_tags_and_date_from_commit_or_tag(
+    /// the commit hash for `git_ref`, and gets any tags for that commit along with the commit date.
+    /// Returns the commit hash, tags, and commit date.  If `git_ref` not a valid git ref, returns
+    /// Error::NotFound.  Returns an error for any other failures
+    pub fn get_commit_and_tags_and_date_from_ref(
         &self,
         software_id: Uuid,
-        commit_or_tag: &str,
+        git_ref: &str,
```

Comment:
Having type aliases for git_refs / git hash/ etc makes sense to me but that's probably a big change...

Reply:
It might not be a huge change, if I'm understanding correctly what you're suggesting, but I don't really understand what the value would be, so I think I'm maybe not understanding correctly.  Can you explain more?

Reply:
If you are passing around hashes in some places and refs in others it can be convenient to have a defined type so that the compiler can catch errors where you pass the wrong one in.

Reply:
Ah okay, that makes sense